### PR TITLE
Added `kubectl create hpa` command

### LINKED ISSUE #39833: Implement `kubectl create autoscaler`
The `kubectl autoscale` command creates and autoscaler, but new creation commands are being standardized as `kubectl create <resource> <name>`.  We should keep the old command, but mirror it either as an alias or new implementation under `kubectl create` so that `kubectl create` is the canonical source for creation commands.

### PR DESCRIPTION
**Pull request by xingzhou**
Added `kubectl create hpa` command, this command share the similar
functionality as `kubectl autoscale` command.

**Which issue this PR fixes**: 
fixes #39833

**Special notes for your reviewer**:
cc @adohe, would you please help to review the code if you have time?

**Release note**:
```
Added `kubectl create hpa` command.
```

### CODE REVIEW

**Comment by k82cn** [comment]
@k8s-bot ok to test

**Comment by fabianofranz** [comment]
@kubernetes/sig-cli-pr-reviews @pwittrock

**Comment by fabianofranz** [comment]
@k8s-bot bazel test this
@k8s-bot verify test this

**Comment by xingzhou** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by adohe** [comment]
@xingzhou sure, I will help review this~

**Comment by xingzhou** [comment]
thanks @adohe and @fabianofranz, have updated the dependencies, now the testings are happy

**Comment by xingzhou** [comment]
Have rebased the code, PTAL, thx

File: pkg/kubectl/cmd/create_hpa.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		kubectl create hpa deployment foo --min=2 --max=10
+
+		# Auto scale a replication controller "foo", with the number of pods between 1 and 5, target CPU utilization at 80%:
+		kubectl create hpa rc foo --max=5 --cpu-percent=80`)
+)
+
+// HPA is a command to ease creating Horizontal Pod Autoscaler.
+func NewCmdCreateHpa(f cmdutil.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "hpa (TYPE NAME | TYPE/NAME) [--min=MINPODS] --max=MAXPODS [--cpu-percent=CPU] [flags]",
```
Comment (kargakis) [inline pkg/kubectl/cmd/create_hpa.go]: We have been creating commands by using the long resource name as the regular name and use the short name as an alias. See [here](https://github.com/kubernetes/kubernetes/blob/e18843d353b21c59b7ebb2b011e9516dcc4fc46d/pkg/kubectl/cmd/create_pdb.go#L48) for example.

**Comment by xingzhou** [comment]
Updated the patch to use long resource name

**Pull request closed without merging**